### PR TITLE
fix: prompt when parameter options are incompatible

### PR DESCRIPTION
**Pull request by mtojek**
Related: https://github.com/coder/coder/issues/9163

This PR modifies CLI and site to prompt user for correct parameter options if they are incompatible now (new template version submitted).

### CODE REVIEW

File: site/src/api/api.ts [hunk 1]
```diff
@@ -1243,6 +1242,35 @@ const getMissingParameters = (
     missingParameters.push(parameter)
   }
 
+  // Check if parameter "options" changed and we can't use old build parameters.
```
Comment (mtojek) [inline site/src/api/api.ts:1245]: @BrunoQuaresma I may need your frontend wisdom. I battling the following edge case here:

0. We have a `foobar` string parameter with options: `aaa`, `bbb` (default), `ccc`, and the workspace built on it. The workspace uses `bbb` option.
1. Change the parameter option `bbb` to something else, let's say `ddd`.
2. Click "Update".

I can see the UpdateParametersDialog, but when I click the `Update` button, it gets back to it. If I pick something else, and then click back to the default, it works fine.

I looked deeper, and it seems that Formik doesn't set the internal `form.values`, even though `initialValues` are passed. I'm not sure how I can proceed here.

File: cli/root.go [hunk 2]
```diff
@@ -981,6 +981,8 @@ func (p *prettyErrorFormatter) format(err error) {
 		msg = sdkError.Message
 		if sdkError.Helper != "" {
 			msg = msg + "\n" + sdkError.Helper
+		} else if sdkError.Detail != "" {
```
Comment (mtojek) [inline cli/root.go:984]: for reviewers:

Message: "Unable to validate parameter X"
Detail: "parameter value must match one of options"

^ this should be consistent with site UI error

File: codersdk/provisionerdaemons.go [hunk 3]
```diff
@@ -69,7 +69,6 @@ const (
 type JobErrorCode string
 
 const (
-	MissingTemplateParameter  JobErrorCode = "MISSING_TEMPLATE_PARAMETER"
```
Comment (mtojek) [inline codersdk/provisionerdaemons.go:72]: for reviewers: used by old legacy parameters (do not exist anymore)

File: site/src/api/api.ts (hunk 1 above)
Comment (mtojek) [reply site/src/api/api.ts:1250]: Apparently, I got it fixes with `enableReinitialize`. Thanks for the suggestion!

**Review by johnstcn — APPROVED** [review]

File: cli/update_test.go [hunk 4]
```diff
@@ -593,12 +593,72 @@ func TestUpdateValidateRichParameters(t *testing.T) {
 			assert.NoError(t, err)
 		}()
 
+		pty.ExpectMatch("Planning workspace...")
+		<-doneChan
+	})
+
+	t.Run("ParameterOptionChanged", func(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (mtojek) [reply cli/update_test.go:600]: Added!

**Review by johnstcn — APPROVED** [review]

**Pull request merged**